doc/manual/src/language/import-from-derivation.md

```diff
@@ -0,0 +1,94 @@
+# Import From Derivation
+
+Import From Derivation (IFD) is a Nix language pattern where an expression's value depends on the contents of a [store object] produced by a [derivation].
+It is characterised by the fact that the store object has to be [realise]d before the expression can be evaluated.
+
+[store object]: /glossary.md#gloss-store-object
+[derivation]: /glossary.md#gloss-derivation
+[realise]: /glossary.md#gloss-realise
+
+This has performance implications:
+Since evaluation is sequential, each required derivation will also be built sequentially.
+
+Passing a derivation `drv` to any built-in function that reads from the filesystem constitutes Import From Derivation:
+
+- [`import`](./language/builtins.md#builtins-import)` drv`
+- [`builtins.readFile`](./language/builtins.md#builtins-readFile)` drv`
+- [`builtins.readDir`](./language/builtins.md#builtins-readDir)` drv`
+- [`builtins.pathExists`](./language/builtins.md#builtins-pathExists)` drv`
+- [`builtins.filterSource`](./language/builtins.md#builtins-filterSource)` f drv`
+- [`builtins.path`](./language/builtins.md#builtins-path)` { path = drv; }`
+- [`builtins.hashFile`](./language/builtins.md#builtins-hashFile)` t drv`
+- `builtins.scopedImport x drv`
+
+Building during evaluation can be disabled by setting [`allow-import-from-derivation`](../command-ref/conf-file.md#conf-allow-import-from-derivation) to `false`.
+
+## Example
+
+In the following Nix expression, the derivation `drv` produces a file containing `1` after `sleep`ing for 5 seconds.
+
+Since `1` is a valid Nix expression itself, it can be [`import`](./builtins.md#builtins-import)ed.
+That requires reading from the output [store path] of `drv`, which has to be [realise]d before its contents `1` can be evaluated.
+
+[store path]: ../glossary.md#gloss-store-path
+
+```nix
+# IFD.nix
+let
+  drv = derivation {
+    name = "expensive";
+    builder = /bin/sh;
+    args = [ "-c" "${/bin/sleep} 5; echo 1 > $out" ];
+    system = builtins.currentSystem;
+  };
+in import drv
+```
+
+```shellSession
+nix-instantiate IFD.nix --eval --read-write-mode
+```
```

Comment:
This does not work for me, I think the Nix expression is wrong:
```
$ nix-instantiate ifd.nix --eval --read-write-mode
error (ignored): error: end of string reached
error: getting status of '/bin/sleep': No such file or directory
(use '--show-trace' to show detailed location information)
```

Indeed this line looks wrong:
```
args = [ "-c" "${/bin/sleep} 5; echo 1 > $out" ];
```

Reply:
Ah, that is because your system does not have `/bin/sleep`. Are you on NixOS?

I would like the manual to be self-sufficient. Importing Nixpkgs and all that stuff would make the example shorter, but also introduce lots of unrelated concepts that readers first have to understand to make sense of the example.

Unfortunately `sleep` is not a POSIX shell builtin, otherwise we could expect the default builder to just work with `sleep 5`. If you have an idea to achieve this with POSIX onboard means, that would be great.

Reply:
Fixed by removing the sleep and instead using a string. It's not important for the principle to focus on the expensive build step.